EXAMPLES.md

```diff
@@ -231,3 +231,33 @@ app.use(auth({
   }
 }));
 ```
+
+## 8. Custom state handling
+
+If your application needs to keep track of the request state before redirecting to log in, you can use the built-in state handling. By default, this library stores the post-callback redirect URL in a state object (along with a generated nonce) that is converted to a string, base64 encoded, and verified during callback (see [our documentation](https://auth0.com/docs/protocols/oauth2/oauth-state) for general information about this parameter). This state object can be added to and used during callback.
+
+You can define a `getLoginState` configuration key set to a function that takes an Express `RequestHandler` and an options object and returned a URL-safe string representation:
+
+```js
+const crypto = require('crypto');
+
+app.use(auth({
+  getLoginState: function (req, options) {
+    const state = {
+      // Property used by the library for redirecting after logging in.
+      returnTo: '/custom-return-path',
+      // Required to make sure the state parameter can't be replayed.
+      nonce: crypto.randomBytes(32).toString('hex');
+      // Additional properties as needed.
+      customProperty: req.someProperty,
+    };
+    // This value will be sent in a URL parameter so it should be transfer-safe.
+    return req.openid.encodeState(state);
```

Comment:
Allowing this for flexibility but could do encoding/decoding in the library automatically ...

Reply:
Are there any use cases where the state should not be encoded? Even if it is a string, it presumably has the potential to contain URL-unsafe characters.

If that holds true I would move the encode operation into the library so that it just always does it and not let the developer hang themselves by doing something that might break the redirect.

Reply:
Thanks Steve. I think you're right, there's just really no reason to require a developer to encode this state if we're managing everything else for them. Will adjust.

Reply:
https://github.com/auth0/express-openid-connect/pull/60/commits/ee5ab85762a62416e3cd97fc4f7ece44fd8fd400